### DISCUSSION TITLE
Lesson 3, cannot use Fund function

### DISCUSSION THREAD

**Discussion by JacobG321** [smartcontractkit/full-blockchain-solidity-course-py#554] [Q&A] [answered]
I am putting more than $50 of wei in to fund it, but it continues to give me this error. Any suggestions?

"Gas estimation errored with the following message (see below). The transaction execution will likely fail. Do you want to force sending?"

```Solidity
// SPDX-License-Identifier: MIT

pragma solidity >=0.6.6 <0.9.0;

import "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
import "@openzeppelin/contracts/utils/math/SafeMath.sol";

contract FundMe {
    using SafeMath for uint256;

    mapping(address => uint256) public addressToAmountFunded;

    function fund() public payable {
      // $50 usd
        uint256 minimumUSD = 50 * 10 ** 18;
        require(getConversionRate(msg.value) >= minimumUSD, "You need to spend more eth!");
        addressToAmountFunded[msg.sender] += msg.value;
        // what the eth -> USD conversion rate

    }

    function getVersion() public view returns (uint256){
        AggregatorV3Interface priceFeed = AggregatorV3Interface(0x8A753747A1Fa494EC906cE90E9f37563A8AF630e);
        return priceFeed.version();
    }

    function getPrice() public view returns (uint256){
        AggregatorV3Interface priceFeed = AggregatorV3Interface(0x8A753747A1Fa494EC906cE90E9f37563A8AF630e);
            (,int256 answer,,,) = priceFeed.latestRoundData();
            return uint256(answer * 1000000000); 

    }
// 1000000000 398811400340000000000000000000

    function getConversionRate(uint256 ethAmount) public view returns (uint256){
         uint256 ethPrice = getPrice();
         uint256 ethAmountInUsd = (ethPrice * ethAmount) / 1000000000000000000;
         return ethAmountInUsd;
}


}
```

**Comment by JacobG321** [comment, marked as answer]
![Untitled](https://user-images.githubusercontent.com/95732210/145810209-dea4cfb3-c41e-444c-8aec-fd6c12787173.png)

**Reply by cromewar** [reply]
Hello @TwoDogsOneBigOneSmall, you are testing this on rinkeby, which means you will have to pay gas fees using the ETH on your account, this is probably failing because you don´t have enough eth to pay for the transaction, or the gas fee is incorrect (too low or to high). If you need more eth please use the [chainlink faucet](https://faucets.chain.link/rinkeby) or optionally paste here your wallet address I'll send you some eth to Rinkeby.

Best regards, Cromewar.

**Reply by JacobG321** [reply]
I have enough ETH, I have .38 in there and I am trying to only use 0.02 ETH which is above minimum that is set in the contract. I have tried more and less gas as well. I am receiving the looping error on all of the functions in this contract, saying it requires an infinite amount of gas.
![Untitled](https://user-images.githubusercontent.com/95732210/145932405-31b641e2-f42e-4e36-bf2b-b01e69309067.png)

**Reply by cromewar** [reply]
Everything seems ok with your code, I reproduced the project to the exact same point you are, and the most likely scenario es you are not sending the correct value to be founded, let's consider this. The minimum entry fee is $50, which translates to this amount of eth:
![image](https://user-images.githubusercontent.com/20199180/146006005-9484a0d8-c40c-4e61-8931-71128e7ac9e2.png)
Using the [eth-coverter](https://eth-converter.com/) this is the amount of wei:
![image](https://user-images.githubusercontent.com/20199180/146006176-e7c083f5-f6fc-41fa-96c7-53915c20728d.png)
So in Remix IDE before sending the Fund transact you must pass the same (better higher) amount of wei as value:
![image](https://user-images.githubusercontent.com/20199180/146006370-45231bb7-a147-48e2-8578-5e5550e9b175.png)
If you set the correct amount when hitting the fund button, it'll work:
![image](https://user-images.githubusercontent.com/20199180/146006482-7aada740-231d-49e2-8683-e0547b279fbf.png)

**Comment by JacobG321** [comment]
Thank you, it went through. I appreciate all your help. I'm sure you'll be seeing more from me soon lol.

**Reply by cromewar** [reply]
Feel free to make any other questions

**Reply by DarrenWayn** [reply]
hey i have the exact problem. i use rinkby testnet and have 1,7 eth, but it continue to fail the fund function. i tried to fund 1 whole eth

**Reply by cromewar** [reply]
remember to manage the gas fees @DarrenWayn

**Comment by DarrenWayn** [comment]
I fix it, for whatever reason the remix needs to be reload
